### DISCUSSION TITLE
Reduce duplication of DTO code

### DISCUSSION THREAD

**Discussion by bartekpacia** [leancodepl/patrol#288] [Ideas]
**TL;DR** Classes representing "commands", such as `TapCommand`, `HandlePermissionCommand`, etc. are written 3 times: in Dart (not exactly, though), Kotlin, and Swift. We should fix that.

Let's take a closer look at a very simple `grantPermissionWhenInUse` method. That's how it's implemented on the Dart side, in `patrol`:

```dart
Future<void> grantPermissionWhenInUse() async {
  await _wrapPost(
    'handlePermission',
    <String, String>{'code': 'WHILE_USING'},
  );
}
```

And in Kotlin:

```kotlin
@Serializable
data class PermissionCommand(var code: String)
```

And in Swift:

```swift
struct PermissionCommand: Codable {
  var code: String
}
```

You can easily see how this can quickly become unmanageable when more `Command`s are added.

I've seen that other new interesting mobile testing tools ([Maestro by mobile.dev](https://github.com/mobile-dev-inc/maestro), [convenient_test](https://pub.dev/packages/convenient_test)) use Protocol Buffers. I'm wondering if we could also use them to generate code in Dart, Swift, and Kotlin from protobuf. 

cc @jakubfijalkowski

**Comment by bartekpacia** [comment]
Done in #338!
